quickwit-core/src/index.rs

```diff
@@ -83,21 +84,23 @@ impl IndexService {
         &self,
         index_config: IndexConfig,
     ) -> Result<IndexMetadata, IndexServiceError> {
-        let index_uri = if let Some(index_uri) = index_config.index_uri.as_ref() {
-            index_uri.to_string()
+        let index_id = index_config.index_id.clone();
+        let index_uri = if let Some(index_uri) = &index_config.index_uri {
+            index_uri.clone()
         } else {
-            let default_index_uri =
-                format!("{}/{}", self.default_index_root_uri, index_config.index_id);
+            let index_uri = self.default_index_root_uri.join(&index_id).expect(
+                "Failed to create default index URI. This should never happen. Please, report.",
+            );
             info!(
-                "`index-uri` is not specified in the index configuration. Setting it to `{}`.",
-                default_index_uri
+                index_id = %index_id,
+                index_uri = %index_uri,
+                "Index config does not specify `index_uri`, falling back to default value.",
             );
-            default_index_uri
+            index_uri
         };
-
         let index_metadata = IndexMetadata {
-            index_id: index_config.index_id.clone(),
-            index_uri,
+            index_id,
+            index_uri: index_uri.into_string(),
```

Comment:
`IndexMetadata.index_uri` should be a `Uri` eventually.

Reply:
Indeed, we could open an issue to track that.